src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Double.cs

```diff
@@ -88,9 +90,55 @@ private void WriteNumberValueIndented(double value)
                 BytesPending += indent;
             }
 
-            bool result = Utf8Formatter.TryFormat(value, output.Slice(BytesPending), out int bytesWritten);
+            bool result = TryFormatDouble(value, output.Slice(BytesPending), out int bytesWritten);
             Debug.Assert(result);
             BytesPending += bytesWritten;
         }
+
+        private static bool TryFormatDouble(double value, Span<byte> destination, out int bytesWritten)
+        {
+            // Frameworks that are not .NET Core 3.0 or higher do not produce roundtrippable strings by
+            // default. Further, the Utf8Formatter on older frameworks does not support taking a precision
+            // specifier for 'G' nor does it represent other formats such as 'R'. As such, we duplicate
+            // the .NET Core 3.0 logic of forwarding to the UTF16 formatter and transcoding it back to UTF8,
+            // with some additional changes to remove dependencies on Span APIs which don't exist downlevel.
+
+#if BUILDING_INBOX_LIBRARY
+            return Utf8Formatter.TryFormat(value, destination, out bytesWritten);
+#else
+            const string FormatString = "G17";
+
+            string utf16Text = value.ToString(FormatString, CultureInfo.InvariantCulture);
+
+            // Copy the value to the destination, if it's large enough.
+
+            if (utf16Text.Length > destination.Length)
+            {
+                bytesWritten = 0;
+                return false;
+            }
+
+            try
+            {
+                byte[] bytes = Encoding.UTF8.GetBytes(utf16Text);
```

Comment:
In .NET Core 2.1, where ToString was also used, this byte[] was avoided by just doing a manual loop:
https://github.com/dotnet/coreclr/pull/22011/files#diff-06426f626f879bee5a9552a7e3abff25R128